backend/app/main.py

```diff
@@ -0,0 +1,147 @@
+    client_kwargs={
+        "scope": conf.OIDC_SCOPES,
+        "timeout": Timeout(timeout=conf.OIDC_REQUEST_TIMEOUT),
+    },
+)
+
+
+@router.get("/login")
+async def login(request: Request):
+    return await oauth.login_portal.authorize_redirect(
+        request=request,
+        redirect_uri=request.url_for("auth"),
+        acr_values=conf.OIDC_ACR_VALUES,
+    )
+
+
+@router.get("/auth")
+async def auth(request: Request):
+    try:
+        token = await oauth.login_portal.authorize_access_token(request)
+    except OAuthError as error:
+        return HTMLResponse(f"<h1>{error.error}</h1>")
+
+    id_token = token["id_token"]
+    access_token = token["access_token"]
+    expires_in = token["expires_in"]
+
+    response = RedirectResponse(url=conf.FRONTEND_URL)
+
+    if id_token:
+        response.set_cookie(
+            key="id_token",
+            value=id_token,
+            max_age=expires_in,
+            httponly=True,
+        )
+    if access_token:
+        response.set_cookie(
+            key="access_token",
+            value=access_token,
+            max_age=expires_in,
+            httponly=True,
+        )
+
+    return response
+
+
+@router.get("/logout")
+async def logout(request: Request, id_token: Optional[str] = Cookie(default=None)):
+    metadata = await oauth.login_portal.load_server_metadata()
+    end_session_endpoint = metadata["end_session_endpoint"]
+
+    end_session_uri = add_params_to_uri(
+        end_session_endpoint,
+        (
+            ("id_token_hint", id_token),
+            ("post_logout_redirect_uri", conf.FRONTEND_URL),
+        ),
+    )
+
+    response = RedirectResponse(url=end_session_uri)
+
+    response.delete_cookie(
+        key="id_token",
+        httponly=True,
+    )
+    response.delete_cookie(
+        key="access_token",
+        httponly=True,
+    )
+
+    return response
+
+
+@router.get("/me")
+def user_profile(id_token: Optional[str] = Cookie(default=None)):
+    """
+    Return information about the currently authenticated user
+    """
+    if id_token:
+        token = jwt.decode(id_token, options={"verify_signature": False})
+        return {
+            "loggedIn": True,
+            "email": token["email"],
+        }
+    return {
+        "loggedIn": False,
+    }
+
+
+@router.post("/data-product/{data_product:path}")
+async def fetch_data_product(
+    data_product: str,
+    request: Request,
+    source=Query(),
+    id_token: Optional[str] = Cookie(default=None),
+):
+    """
+    Simple proxy from frontend to Product Gateway, because frontend
+    can't access Product Gateway directly due to CORS
+    """
+    body = await request.json()
+    headers = {}
+    if id_token:
+        headers["authorization"] = f"Bearer {id_token}"
+
+    resp = httpx.post(
+        f"{conf.PRODUCT_GATEWAY_URL}/{data_product}",
+        params={"source": source},
+        json=body,
+        headers=headers,
+        timeout=30,
```

Comment:
The `timeout` could maybe be a setting rather than a hardcoded value. Maybe one option could be to use the same variable/value for both the login portal and product gateway.